src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ILspEditorFeatureDetector.cs

```diff
@@ -5,7 +5,7 @@ namespace Microsoft.VisualStudio.Razor;
 
 internal interface ILspEditorFeatureDetector
 {
-    bool IsLspEditorEnabled();
+    bool IsLspEditorAvailable(string? documentMoniker);
```

Comment:
Imo, to help with clarity, can we have just two calls? Might help with not having this happen again

```csharp
// Checks the setting
bool IsLspEditorEnabled();

// Checks the capability
bool DoesDocumentSupportLspEditor(string? documentMonikor);
```

Reply:
I wouldn't even mind the aggregate name of `IsLspEditorEnabledAndSupported(string? documentMonikor)` that just calls both of these

Reply:
I think we would always want to check both, so I don't know if splitting in two separate methods makes sense, but I'm on board with being more explicit with naming - renamed to EnabledAndSupported